### DISCUSSION TITLE
Is there a way to expand this div to 100% of the screen without the scrollbar to appear?

### DISCUSSION THREAD

**Discussion by frederikhors** [tailwindlabs/tailwindcss#14763] [Help]
I have this html and tailwindcss code:

```html
<div class="flex flex-col min-h-screen">
    <div class="bg-black text-white h-[65px]">
        This is a navbar, but 65px is just an example, the height is dynamic on each page
    </div>

    <div class="bg-blue-200 flex-1">
        Body here

        <div class="bg-pink-300">
            I would like this to be 100% height of the screen but without the need to scroll the page. If I use `h-screen` the scroll appears.
        </div>
    </div>

    <div class="bg-black text-white h-[65px]">
        This is a footer, but 65px is just an example, the height is dynamic on each page
    </div>
</div>
```

Ideally if dinamically the navbar or the footer change their height the height should change accordingly.

I can fix this using `flex flex-col flex-1` on the parents divs.

But in some pages I can't. Because sometimes I cannot change the parent divs classes.

Obviously I'm not looking for a Javascript solution.

Can you help me?

**Comment by aniruddhaadak80** [comment]
Hi @frederikhors ,

### Issue: Expand Inner Div to 100% Height Without Scrollbar

To make the inner pink div expand to 100% height of the screen without causing a scrollbar, you can use a combination of CSS properties. Here’s how you can achieve this using Tailwind CSS:

1. **Remove Fixed Height:** Avoid using fixed heights for the inner div.

2. **Use `flex-grow`:** Set the pink div to grow and take the remaining space.

3. **Set `overflow` to Hidden:** This prevents scrollbars from appearing.

Here’s how you can adjust your code:

```html
<div class="flex flex-col min-h-screen overflow-hidden">
    <div class="bg-black text-white h-auto">
        This is a navbar, but height is dynamic on each page.
    </div>

    <div class="bg-blue-200 flex-1 flex flex-col overflow-hidden">
        Body here

        <div class="bg-pink-300 flex-1 overflow-hidden">
            I would like this to be 100% height of the screen but without the need to scroll the page.
        </div>
    </div>

    <div class="bg-black text-white h-auto">
        This is a footer, but height is dynamic on each page.
    </div>
</div>
```

### Key Changes:
- **`h-auto` on Navbar and Footer:** This allows them to grow based on their content.
- **`flex-1` on the pink div:** This allows it to occupy the remaining space between the navbar and footer.
- **`overflow-hidden`:** This is added to prevent scrollbars from appearing due to overflow.

This setup should allow your pink div to occupy the available height dynamically without triggering a scrollbar. Let me know if you have any questions!

**Reply by frederikhors** [reply]
This is a typical ChatGPT answer! Congratulations! And it doesn't work if I add an empty div: https://play.tailwindcss.com/DHMZREIVmX.

**Reply by aniruddhaadak80** [reply]
Hey @frederikhors ,

Try this out ⬇️⬇️⬇️⬇️
```
<div class="flex flex-col h-screen">
    <div class="bg-black text-white h-[65px]">
        This is a navbar, but 65px is just an example, the height is dynamic on each page
    </div>

    <div class="bg-blue-200 flex-1 overflow-hidden">
        Body here

        <div class="bg-pink-300 h-full">
            I will take full remaining height without scrollbars
        </div>
    </div>

    <!-- Your new div can go here without causing scroll -->
    <div class="bg-red-200">
        New content here
    </div>

<!-- You can add as many div as you want now -->
    <div>You can add as many div as you want now </div>
    <div>You can add as many div as you want now </div>
    <div>You can add as many div as you want now </div>
    <div>You can add as many div as you want now </div>
    <div>You can add as many div as you want now </div>
    <div>You can add as many div as you want now </div>



    <div class="bg-black text-white h-[65px]">
        This is a footer, but 65px is just an example, the height is dynamic on each page
    </div>
</div>

```
https://play.tailwindcss.com/BdpGavOGJw

Thank you .

**Reply by frederikhors** [reply]
Please stop using AI (not I).

**Comment by UtkarshK95** [comment]
To make the inner div take up the full remaining height of the screen without causing a scrollbar, you can use CSS’s `calc()` function combined with Tailwind’s utility classes. Here’s how to adjust your structure and classes:

```html
<div class="flex flex-col min-h-screen">
    <div class="bg-black text-white h-[65px]">
        This is a navbar, but 65px is just an example, the height is dynamic on each page
    </div>

    <div class="bg-blue-200 flex-1 relative">
        Body here

        <div class="bg-pink-300 absolute top-0 left-0 w-full" style="height: calc(100vh - 130px);">
            I would like this to be 100% height of the screen but without the need to scroll the page.
        </div>
    </div>

    <div class="bg-black text-white h-[65px]">
        This is a footer, but 65px is just an example, the height is dynamic on each page
    </div>
</div>
```

### Explanation:

1. **Set `relative` on the parent container** of the pink div (`bg-blue-200`) so that the inner pink div (`bg-pink-300`) can position itself within it.
  
2. **Use `absolute` positioning** for the pink div (`bg-pink-300`) and set `top-0 left-0 w-full` to make it take up the full width, starting from the top of its parent container.

3. **Calculate Height Using `calc(100vh - 130px);`:** Adjust the pink div's height dynamically by subtracting the combined height of the navbar and footer (in this case, `65px + 65px = 130px`) from `100vh`. Adjust the `130px` accordingly if navbar or footer heights change dynamically.

**Reply by frederikhors** [reply]
This is a typical ChatGPT answer! Congratulations! And the footer disappears if I add an empty div: https://play.tailwindcss.com/DHMZREIVmX.

**Comment by wongjn** [comment]
You could look at applying CSS to the parents through `:has(&)` variant, like https://play.tailwindcss.com/8jNs8d4qAU
